packages/test/src/test-patch-and-condition.ts

```diff
@@ -14,6 +14,7 @@ if (RUN_INTEGRATION_TESTS) {
     const worker1 = await Worker.create({
       taskQueue: 'patch-in-condition',
       workflowsPath: require.resolve('./workflows/patch-and-condition-pre-patch'),
+      maxCachedWorkflows: 0,
```

Comment:
Curious why you set this..

Reply:
This fixes a very subtile failure introduced by this [apparently innocuous PR](https://github.com/temporalio/sdk-core/commit/c72322f4ae317f0745f65e89fa4c5912ba0e2498)...

Essentially, once worker1 get stopped, WFT will still be sent to worker1's sticky queue. Before, worker2 would register to the very same sticky queue and thus, would _immediately_ be able to accept the WFT, without delay. But now that sticky queue names are randomized, it the WFT will have to wait 10 seconds in the sticky queue before it gets moved to the regular queue so that it can be picked up by worker2. And then the same goes for worker2 to worker3.

Now, the workflow use a 5 second timer, and then dies. That timer still tick even while the WFT is waiting in the queue, so the workflow will now completes in worker2, rather than staying alive until worker3.

I could have avoid this by increasing the sleep time, but that would have increased the total execution time of this test. By disabling caching, we both resolve the issue without side effect on execution time.